src/v/model/metadata.h

```diff
@@ -203,6 +204,7 @@ struct broker_shard {
     uint32_t shard;
     friend std::ostream& operator<<(std::ostream&, const broker_shard&);
     bool operator==(const broker_shard&) const = default;
+    auto operator<=>(const model::broker_shard&) const = default;
```

Comment:
Since we want to sort broker_shards in the test (to canonicalize them), add a spaceship here.